src/contracts/atlas/Escrow.sol

```diff
@@ -345,22 +346,13 @@ abstract contract Escrow is AtlETH {
             }
         }
 
-        // If there are no errors, attempt to execute
-        if (!result.canExecute() || !_trySolverLock(solverOp)) return 0;
-
-        data = abi.encodeCall(
-            IExecutionEnvironment.solverMetaTryCatch,
-            (IDAppControl(dConfig.to).getBidValue(solverOp), gasLimit, solverOp, data)
-        );
-
-        data = abi.encodePacked(data, key.holdSolverLock(solverOp.solver).pack());
-
-        (success, data) = key.executionEnvironment.call{ value: solverOp.value }(data);
-
-        _releaseSolverLock(solverOp, gasWaterMark, result, true, true);
+        data = abi.encodeCall(IAtlas.solverCall, (solverOp.bidAmount, gasLimit, key, solverOp, data));
+        (success, data) = address(this).call(data);
 
+        // The `solverCall()` above should always revert as key.bidFind is always true when it's called in the context
+        // of this function. Therefore `success` should always be false below, and the revert should be unreachable.
         if (success) {
-            revert();
+            revert Unreachable();
         }
 
```

Comment:
Is this just being left in here for extra safety? We should be able to make a test that asserts this is true which could allow us to remove this code and save some gas?

```suggestion
        if (success) {
            revert Unreachable();
        }
```

Reply:
Yeah the auditors had a note that even though its unreachable, because its so non-standard we should be quite clear about whats happening here, so there's the comment and a named error to communicate it.

We could remove it and have a test, but it would only save a tiny amount of gas (60 gas*) and size (31 bytes), and it feels more valuable to keep it in `_getBidAmount()` to communicate that we're doing things in a pretty non-standard way, and its there as an extra safety measure that should never be needed.

* Diff from metacall tx cost in `test_ExPostOrdering`